scipy/signal/_splinemodule.h

```diff
@@ -1,25 +1,340 @@
+#pragma once
 #include "Python.h"
-#include "numpy/arrayobject.h"
-#include <math.h>
 
+#ifndef M_PI
+#define M_PI           3.14159265358979323846  /* pi */
+#endif
 
-#define PYERR(message) do {PyErr_SetString(PyExc_ValueError, message); goto fail;} while(0)
+#ifndef PyArray_MIN
+#define PyArray_MIN(a,b) (((a)<(b))?(a):(b))
+#endif
 
-static void convert_strides(npy_intp*,npy_intp*,int,int);
+#define PYERR(message)                                                                                                 \
+    do {                                                                                                               \
+        PyErr_SetString(PyExc_ValueError, message);                                                                    \
+        goto fail;                                                                                                     \
+    } while (0)
 
-extern int S_SYM_IIR1_initial(float, float*, float*, int, int, float);
-extern int S_SYM_IIR2_initial_fwd(double, double, float*, float*, int, int, float);
-extern int S_SYM_IIR2_initial_bwd(double, double, float*, float*, int, int, float);
-extern int S_separable_2Dconvolve_mirror(float*,float*,int,int,float*,float*,int,int,npy_intp*,npy_intp*);
 
-extern int D_SYM_IIR1_initial(double, double*, double*, int, int, double);
-extern int D_SYM_IIR2_initial_fwd(double, double, double*, double*, int, int, double);
-extern int D_SYM_IIR2_initial_bwd(double, double, double*, double*, int, int, double);
-extern int D_separable_2Dconvolve_mirror(double*,double*,int,int,double*,double*,int,int,npy_intp*,npy_intp*);
+template <typename T, typename C>
+int _sym_iir1_initial(C z1, C *x, C *yp0, int M, int N, T precision) {
```

Comment:
```suggestion
/**
Compute the starting initial conditions for the system
1 / (1 - z1 * z^-1) against signal x.

The initial conditions are defined as x[0] + z1 * \sum{k = 0}^{n - 1} x[k] * z1^k
this sum will be aggregated until its precision is below a certain threshold.

Arguments
----------------
z1: double or complex
   Exponential decay parameter in the transfer function
x: float* or double* or complex*
   2D strided pointer signal of size (M, N). When M > 1, multiple signals
    will be processed independently.
yp0: float* or double* or complex*
    Output state condition pointer of size (M, 1)
M: int
    Number of signals to compute initial conditions for.
N: int
    Length of the signals.
precision: double* or float*
    Precision up to which the initial conditions will be computed.
**/
int _sym_iir1_initial(C z1, C *x, C *yp0, int M, int N, T precision) {
```

Reply:
Thank you Edgar! Re-applied your edit which somehow got lost in one of rebases.